scripts/diff_against_branch.sh

```diff
@@ -34,71 +44,70 @@ function check_added_file {
-	echo
+	echo -e "🆕 Detected newly added file: ${PURPLE}$FILE${NC}" 
+	echo -e "🔢 Version: ${PURPLE}$FILE_VERSION${NC}" 
+	echo -e "🔃 Comparing against ${PURPLE}$ORIGIN/$OTHER_FILE${NC}"
 
 	GIT_PAGER=cat git diff --exit-code origin/$ORIGIN:$OTHER_FILE $FILE
 	if [[ "$?" == "0" ]] ; then
-		echo "## No change detected! (weird?)"
+		echo "❓ No change detected! (weird?)"
 	fi
+	echo
 }
 
 function check_modified_file {
 	FILE=$1
 	OTHER_FILE=$FILE
 
 	echo
-	echo "#############################################################################"
-	echo "## Detected modified file: $FILE" 
-	echo "## Comparing against $ORIGIN/$OTHER_FILE"
-	echo "#############################################################################"
-	echo
+	echo -e "🔧 Detected modified file: ${PURPLE}$FILE${NC}" 
+	echo -e "🔃 Comparing against ${PURPLE}$ORIGIN/$OTHER_FILE${NC}"
 
 	GIT_PAGER=cat git diff --exit-code origin/$ORIGIN:$OTHER_FILE $FILE
 	if [[ "$?" == "0" ]] ; then
-		echo "## No change detected! (weird?)"
+		echo "❓ No change detected! (weird?)"
 	fi
 
 	# Check here if the file has any structure modifications, if yes return
 	# some value != 0 for an automated script to fail if we detect any modifications
 	# against the c4t branch	
 	diff -w -B - <(git show origin/$ORIGIN:$OTHER_FILE | sed -e "s#//.*##g") < <(cat $FILE | sed -e "s#//.*##g") > /dev/null
 	if [[ "$?" != "0" ]] ; then
-		echo
-		echo "## ❌ [FAIL] ERROR: Structural change detected in already existing file: $FILE"
+		echo -e "❌ ${RED}[FAIL] ERROR${NC}: Structural change detected in already existing file: ${PURPLE}$FILE${NC}"
 		ERROR_FILES+=("$FILE")
 	else
-		echo
-		echo "## ✅ [PASS] INFO: No structural change detected. The changes should only be in the comments."
+		echo -e "✅ ${GREEN}[PASS] INFO${NC}: No structural change detected. The changes should only be in the comments."
 	fi 
+	echo
 }
 
+echo -e "⏳ Fetching the latest changes from the origin branch: ${PURPLE}$ORIGIN${NC}"
+git fetch origin $ORIGIN > /dev/null 2>&1
+
+echo -e "🔍 Checking for illegal filesystem changes like removing/moving existing files"
 while read FILE ; do
-	check_added_file $FILE
-done < <(git diff --name-status origin/$ORIGIN | grep -P "^A.*" | grep -oP "proto/.*")
+	check_filesystem_changes $FILE
+done < <(git diff --diff-filter=am --name-status origin/$ORIGIN | grep -oP "proto/.*")
 
+echo -e "🔍 Checking for added files to print out the diff of the new version against the previous version"
+while read FILE ; do
+	check_added_file $FILE
+done < <(git diff --diff-filter=A --name-status origin/$ORIGIN | grep -oP "proto/.*")
 
+echo -e "🔍 Checking for modifications in existing files to catch unwanted structural changes"
 while read FILE ; do
 	check_modified_file $FILE
-done < <(git diff --name-status origin/$ORIGIN | grep -P "^M.*" | grep -oP "proto/.*")
+done < <(git diff --diff-filter=M --name-status origin/$ORIGIN | grep -P "^M.*" | grep -oP "proto/.*")
```

Comment:
I think no need for the first grep here

```suggestion
done < <(git diff --diff-filter=M --name-status origin/$ORIGIN | grep -oP "proto/.*")
```

Reply:
Good catch! Fixed ✅